etc/vendor-libmongoc.sh

```diff
@@ -3,8 +3,14 @@
 set -eou pipefail
 
 PWD=`pwd`
-LIBMONGOC_VERSION=1.18.0
-TARBALL_URL=https://github.com/mongodb/mongo-c-driver/releases/download/$LIBMONGOC_VERSION/mongo-c-driver-$LIBMONGOC_VERSION.tar.gz
+LIBMONGOC_MAJOR_VERSION=1
+LIBMONGOC_MINOR_VERSION=18
+LIBMONGOC_PATCH_VERSION=0
+LIBMONGOC_PRERELEASE_VERSION=
+
+LIBMONGOC_FULL_VERSION=${LIBMONGOC_MAJOR_VERSION}.${LIBMONGOC_MINOR_VERSION}.${LIBMONGOC_PATCH_VERSION}${LIBMONGOC_PRERELEASE_VERSION:+-$LIBMONGOC_PRERELEASE_VERSION}
```

Comment:
what does the `:+-` syntax do here?

Reply:
if the value before the colon is set, it expands to whatever comes after the colon. if the value before the colon is unset, it expands to nothing.

so basically, if the prerelease version is set, the string becomes "-[prerelease version here]" e.g. "-alpha2", but if the prerelease version is not set this is just empty. 

this was necessary so we only append a dash if there is a prerelease version 

(thank you stack overflow for this: https://stackoverflow.com/a/40771884)

Reply:
I guess you could also think of it like a ternary statement: `${a:+b}` is equivalent to `a != nil ? b : nil`

Reply:
ah makes sense, thanks for the explanation :+1: